iModelCore/ECDb/Tests/NonPublished/ProfileTests.cpp

```diff
@@ -360,4 +365,490 @@ TEST_F(ECDbTestFixture, ImportSchemaByProfileVersion)
+        <ECCustomAttributeClass typeName="Foo" modifier="Sealed" appliesTo="CustomAttributeClass">
+        </ECCustomAttributeClass>
+
+        <ECCustomAttributeClass typeName="Bar" modifier="Sealed" appliesTo="EntityClass">
+            <ECCustomAttributes>
+                <Foo></Foo>
+            </ECCustomAttributes>
+        </ECCustomAttributeClass>
+        <ECEntityClass typeName="MyEntity" >
+            <ECCustomAttributes>
+                <Bar></Bar>
+            </ECCustomAttributes>
+            <ECProperty propertyName="Length" typeName="double" />
+        </ECEntityClass>
+
+        <ECEntityClass typeName="MySubclass" >
+            <BaseClass>MyEntity</BaseClass>
+        </ECEntityClass>
+        </ECSchema>)xml");
+
+    ASSERT_EQ(BentleyStatus::SUCCESS, ImportSchema(schema));
+    {
+    ECIssueListener issueListener(m_ecdb);
+
+    ECSqlStatement stmt;
+    ASSERT_EQ(ECSqlStatus::Success, stmt.Prepare(m_ecdb, "SELECT * from s1.MySubclass"));
+
+    auto lastIssue = issueListener.GetIssue();
+    ASSERT_FALSE(lastIssue.has_value()) << "Should not raise an issue.";
+    }
+    
+    SchemaItem schema2(R"xml(<?xml version="1.0" encoding="utf-8" ?>
+        <ECSchema schemaName="Schema1" alias="s1" version="1.0.2" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2">
+        <ECSchemaReference name="ECDbMap" version="02.00.02" alias="ecdbmap"/>
+        <ECCustomAttributeClass typeName="Foo" modifier="Sealed" appliesTo="CustomAttributeClass">
+            <ECCustomAttributes>
+                <UseRequiresVersion xmlns="ECDbMap.02.00.02">
+                    <ECDbRuntimeVersion>999.9.9.9</ECDbRuntimeVersion>
+                </UseRequiresVersion>
+            </ECCustomAttributes>
+        </ECCustomAttributeClass>
+
+        <ECCustomAttributeClass typeName="Bar" modifier="Sealed" appliesTo="EntityClass">
+            <ECCustomAttributes>
+                <Foo></Foo>
+            </ECCustomAttributes>
+        </ECCustomAttributeClass>
+        <ECEntityClass typeName="MyEntity" >
+            <ECCustomAttributes>
+                <Bar></Bar>
+            </ECCustomAttributes>
+            <ECProperty propertyName="Length" typeName="double" />
+        </ECEntityClass>
+
+        <ECEntityClass typeName="MySubclass" >
+            <BaseClass>MyEntity</BaseClass>
+        </ECEntityClass>
+        </ECSchema>)xml");
+    
+    ASSERT_EQ(BentleyStatus::SUCCESS, ImportSchema(schema2));
+
+    {
+    ECIssueListener issueListener(m_ecdb);
+
+    ECSqlStatement stmt;
+    ASSERT_EQ(ECSqlStatus::InvalidECSql, stmt.Prepare(m_ecdb, "SELECT * from s1.MySubclass"));
+
+    auto lastIssue = issueListener.GetIssue();
+    ASSERT_TRUE(lastIssue.has_value()) << "Should raise an issue.";
+    ASSERT_STREQ("Invalid ECClass in ECSQL: Cannot use ECClass 'Schema1:MySubclass' because it requires a newer version of ECDb.", lastIssue.message.c_str());
+    }
+
+    CloseECDb();
+    }
+

```

Comment:
how about on struct and relationship class?
How about entity class or relationship class as endpoint of relationship class?
CAs on relationship constraints

Reply:
Added tests